runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/middleware/RetryMiddleware.kt

```diff
@@ -47,6 +51,10 @@ internal class RetryMiddleware<I, O>(
                 withSpan<RetryMiddleware<*, *>, _>("Attempt-$attempt") {
                     if (attempt > 1) {
                         coroutineContext.debug<RetryMiddleware<*, *>> { "retrying request, attempt $attempt" }
+                        when (strategy::class) {
+                            StandardRetryStrategy::class -> modified.context.emitBusinessMetric(BusinessMetrics.RETRY_MODE_STANDARD)
+                            AdaptiveRetryStrategy::class -> modified.context.emitBusinessMetric(BusinessMetrics.RETRY_MODE_ADAPTIVE)
+                        }
```

Comment:
correctness: this is only emitting metrics when a request is retried (first attempt failed), but it seems like the SEP wants us to emit the metrics any time the strategy is used (even on the first attempt).